pyro/ops/integrator.py

```diff
@@ -44,3 +34,38 @@ def _grad(potential_fn, z):
     grads = grad(potential_fn(z), z_nodes)
     grads = [v.data for v in grads]
     return dict(zip(z_keys, grads))
+
+
+def velocity_verlet(z, r, potential_fn, step_size, num_steps, transforms=None):
+    """
+    Second order symplectic integrator that uses the velocity verlet algorithm.
+
+    :param dict z: dictionary of sample site names and their current values
+        (type ``torch.autograd.Variable``).
+    :param dict r: dictionary of sample site names and corresponding momenta
+        (type ``torch.autograd.Variable``).
+    :param callable potential_fn: function that returns potential energy given z
+        for each sample site. The negative gradient of the function with respect
+        to ``z`` determines the rate of change of the corresponding sites'
+        momenta ``r``.
+    :param float step_size: step size for each time step iteration.
+    :param int num_steps: number of discrete time steps over which to integrate.
+    :param dict transforms: Optional dictionary that specifies a transform
+        for a sample site with constrained support to unconstrained space. The
+        transform should be invertible, and implement `log_abs_det_jacobian`.
+    :return tuple (z_next, r_next): final position and momenta, having same types as (z, r).
+    """
+    if transforms is None:
```

Comment:
I kind of liked the `transoforms={}` which would make this check become `if not transforms: ...`. Do you have a reason for preferring None?

Reply:
Using `{}` is fine since we are not mutating the default argument. My linter just gives me a warning when we use a mutable default argument, so I tend to not use them, even though the correct null representation for dict types should be an empty dict. Will change.

Reply:
OK, thanks.

And stop working on your vacation :smile: